SU2_CFD/src/transfer_structure.cpp

```diff
@@ -137,46 +144,76 @@ void CTransfer::Scatter_InterfaceData(CSolver *donor_solution, CSolver *target_s
     for (iMarkerDonor = 0; iMarkerDonor < nMarkerDonor; iMarkerDonor++) {
       /*--- If the tag GetMarker_All_FSIinterface(iMarkerDonor) equals the index we are looping at ---*/
       if ( donor_config->GetMarker_All_FSIinterface(iMarkerDonor) == iMarkerInt ) {
-        /*--- We have identified the local index of the Donor marker ---*/
-        /*--- Store the number of local points that belong to Marker_Donor on each processor ---*/
-        /*--- This are the number of points that will be sent from this particular processor ---*/
-        /*--- This includes the halo nodes ---*/
-        nLocalVertexDonor = donor_geometry->GetnVertex(iMarkerDonor);
-        /*--- Store the identifier for the structural marker ---*/
         Marker_Donor = iMarkerDonor;
         /*--- Exit the for loop: we have found the local index for iMarkerFSI on the FEA side ---*/
         break;
       }
-      else {
-        /*--- If the tag hasn't matched any tag within the donor markers ---*/
-        nLocalVertexDonor = 0;
-        Marker_Donor = -1;
-      }
     }
     
     /*--- On the target side ---*/
     
     for (iMarkerTarget = 0; iMarkerTarget < nMarkerTarget; iMarkerTarget++) {
       /*--- If the tag GetMarker_All_FSIinterface(iMarkerFlow) equals the index we are looping at ---*/
       if ( target_config->GetMarker_All_FSIinterface(iMarkerTarget) == iMarkerInt ) {
-        /*--- We have identified the local index of the Flow marker ---*/
-        /*--- Store the number of local points that belong to Marker_Flow on each processor ---*/
-        /*--- This includes the halo nodes ---*/
-        nLocalVertexTarget = target_geometry->GetnVertex(iMarkerTarget);
-        /*--- Store the identifier for the fluid marker ---*/
         Marker_Target = iMarkerTarget;
         /*--- Exit the for loop: we have found the local index for iMarkerFSI on the FEA side ---*/
         break;
       }
-      else {
-        /*--- If the tag hasn't matched any tag within the Flow markers ---*/
-        nLocalVertexTarget = 0;
-        Marker_Target = -1;
+    }
+
+    #ifdef HAVE_MPI
+
+    Donor_check  = -1;
+    Target_check = -1;
+
+    /*--- We gather a vector in MASTER_NODE that determines if the boundary is not on the processor because of the partition or because the zone does not include it  ---*/
```

Comment:
Yeah, no limitation on the number of zones, at least for fluid-fluid problems which are the ones I tested properly, but it should work also for structure-structure problems.
I'm not sure about fsi cause that driver includes 2 mixed zones (assuming that the first is fluid and the second structure, or the other way around). 
With multiple fsi drivers we'll have 2 layers of communication, one between fsi drivers and the other between fluid and structure within each driver.
We should modify first the config file so that we can specify which solver has to manage each zone (and the kind of interpolation for each interface) and which zones are coupled by fsi driver. 
We'll have to think a way of doing this in the most general way so to have the possibility to run every possible combination of solvers but yeah, once this issue is solved I think it will be straightforward.